x/stakeibc/module_ibc.go

```diff
@@ -211,7 +186,40 @@ func (im IBCModule) NegotiateAppVersion(
 	return proposedVersion, nil
 }
 
+func (im IBCModule) UnmarshalAck(ctx sdk.Context, acknowledgement []byte) (*channeltypes.Acknowledgement_Result, error) {
+	ack := channeltypes.Acknowledgement_Result{}
+	eventType := "callback"
+	ctx.EventManager().EmitEvent(
+		sdk.NewEvent(
+			eventType,
+			sdk.NewAttribute(sdk.AttributeKeyModule, types.ModuleName),
+			sdk.NewAttribute(types.AttributeKeyAck, fmt.Sprintf("%v", ack)),
+		),
+	)
+	err := json.Unmarshal(acknowledgement, &ack)
+	if err != nil {
+		ackErr := channeltypes.Acknowledgement_Error{}
+		err := json.Unmarshal(acknowledgement, &ackErr)
+		if err != nil {
+			ctx.EventManager().EmitEvent(
+				sdk.NewEvent(
+					eventType,
+					sdk.NewAttribute(types.AttributeKeyAckError, ackErr.Error),
+				),
+			)
+			im.keeper.Logger(ctx).Error("Unable to unmarshal acknowledgement error", "error", err, "data", acknowledgement)
+			return nil, err
+		}
+		im.keeper.Logger(ctx).Error("Unable to unmarshal acknowledgement result", "error", err, "remote_err", ackErr, "data", acknowledgement)
+		return nil, err
```

Comment:
nit: logging bytes. Also maybe rephrase the second log. If I'm understanding correctly, this isn't really an unmarshalling error, it's just that ack itself was an error response